### ISSUE TITLE
Enhancement: Replace DEBUG_LINT with use of __attribute__((format(printf ...)))

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/mkiocccentry#21]
The need for a DEBUG_LINT macro set is a holdover from back in the late 1980's.

We can replace the need for DEBUG_LINT by using `__attribute__((format(printf ...)))` instead.  For example:

```c
extern void msg(const char *fmt, ...) \
        __attribute__((format(printf, 1, 2)));          /* 1=format 2=params */
extern void dbg(int level, const char *fmt, ...) \
        __attribute__((format(printf, 2, 3)));          /* 2=format 3=params */
extern void warn(const char *name, const char *fmt, ...) \
        __attribute__((format(printf, 2, 3)));          /* 2=format 3=params */
extern void warnp(const char *name, const char *fmt, ...) \
        __attribute__((format(printf, 2, 3)));          /* 2=format 3=params */
extern void err(int exitcode, const char *name, const char *fmt, ...) \
        __attribute__((noreturn)) __attribute__((format(printf, 3, 4)));        /* 3=format 4=params */
extern void errp(int exitcode, const char *name, const char *fmt, ...) \
        __attribute__((noreturn)) __attribute__((format(printf, 3, 4)));        /* 3=format 4=params */
extern void vfprintf_usage(int exitcode, FILE *stream, const char *fmt, ...) \
         __attribute__((format(printf, 3, 4)));                                 /* 3=format 4=params */
```

We are aware of the formatted error messages err() and friends.  That question about their use will appear in a later enhancement issue.

Comments and questions welcome.

**Comment by lcn2** [comment]
Resolved with the release of version 0.30 2022-01-21.

**Comment by ilyakurdyukov** [comment]
MSVC, TenDRA and Little C Compiler doesn't support `__attribute__`. TCC support or ignore.

**Comment by lcn2** [comment]
Good idea, @ilyakurdyukov thanks.

Fixed in 31d783705883c1093c9d355e836e7d3d85da7031
